homework 1/ReaderWriterLock/SharedResourcePerfomanceTests.cs

```diff
@@ -14,30 +16,51 @@ public class SharedResourcePerformanceTests
     private const int WritersThreads = 100;
     private const int ReadersThreads = 1000;
     private const int NumberOfIterations = 10000;
-    private const int FactorialNumber = 60; // Большое число для вычисления факториала
+    private const int FactorialNumber = 500;
 
     [Test]
     public void TestLockPerformance()
     {
         _sharedResource = new SharedResourceLock();
-        long lockTime = MeasurePerformance();
+        var lockTime = MeasurePerformance();
         Console.WriteLine($"Lock time taken: {lockTime} ms");
 
         _sharedResource = new SharedResourceRwLock();
-        long rwLockTime = MeasurePerformance();
+        var rwLockTime = MeasurePerformance();
         Console.WriteLine($"ReaderWriterLock time taken: {rwLockTime} ms");
 
-        // Проверка, что время выполнения с ReaderWriterLock меньше, чем с Lock
-        ClassicAssert.Less(rwLockTime, lockTime, "ReaderWriterLock should be faster than Lock");
+        rwLockTime.Should().BeLessThan(lockTime);
     }
 
     private long MeasurePerformance()
     {
-        // Нужно реализовать тест производительности.
-        // В многопоточном режиме нужно запустить:
-        // - Чтение общего ресурса в количестве ReadersThreads читающих потоков
-        // - Запись значений в количестве WritersThreads записывающих потоков
-        // - В вызовах читателей и писателей обязательно нужно вызывать подсчет факториала для симуляции полезной нагрузки
-        throw new NotImplementedException();
+        var writers = Enumerable.Range(0, WritersThreads).Select(_ => new Thread(WriteData));
+        var readers = Enumerable.Range(0, ReadersThreads).Select(_ => new Thread(ReadData));
+        var threads = writers.Concat(readers).ToArray();
+        var sw = Stopwatch.StartNew();
+
+        threads.AsParallel().ForAll(t => t.Start());
+        threads.AsParallel().ForAll(t => t.Join());
```

Comment:
задумка запустить потоки одновременно хорошая, но немного странно использовать тредпул для запуска других потоков. можно по-другому: завести `var event = new ManualResetEvent(false)`; в начале каждого писателя/читателя сделать `event.Wait()`; в основном потоке сделать так
```suggestion
        threads.ForEach(t => t.Start());
        event.Set();
        threads.ForEach(t => t.Join());
```
- тогда все потоки начнут выполнять работу одновременно